### PR TITLE
Fix #148.

### LINKED ISSUE #148: `pubspec_overrides.yaml` not supported
**Describe the bug**
Since [this pull request](https://github.com/dart-lang/pub/pull/3215), a separate `pubspec_overrides.yaml` file can exist in a repo. The `diff` tool does not read this file, thus ignoring potential dependencies.

**To Reproduce**
Run this tool on `package:firehose` in a monorepo at https://github.com/dart-lang/ecosystem/tree/main/pkgs/firehose, which has a `pubspec_overrides.yaml` with a local dependency

```yaml
dependency_overrides:
  dart_flutter_team_lints:
    path: ../dart_flutter_team_lints
```

**Actual and Expected behavior**
Some output on the diff.

**System info (please complete the following information):**
 - `Dart SDK version: 3.1.0 (stable) (Tue Aug 15 21:33:36 2023 +0000) on "linux_x64"`
 - `0.15.1-dev`

**Additional context**

### PR DESCRIPTION
**Pull request by mosuem**
## Description
Fix #148 by copying dependency overrides.

Needs to wait for merge of https://github.com/dart-lang/pubspec_parse/pull/109.

## Type of Change

- [ ] 🚀 New feature (non-breaking change)
- [x] 🛠️ Bug fix (non-breaking change)
- [ ] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [ ] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [ ] 📝 Documentation
- [ ] 🧹 Chore / Housekeeping

### CODE REVIEW

**Comment by devmil** [comment]
Nice! Thanks for your contribution. Will have a look once back from vacation 😁

File: pubspec.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@ name: dart_apitool
 description: A tool to analyze the public API of a package, create a model of it and diff it against another version to check semver.
 repository: https://github.com/bmw-tech/dart_apitool
 
-version: 0.15.1-dev
```
Comment (devmil) [inline pubspec.yaml:5]: The "dev" postfix is kept till the next release happens. The release process will remove it and jump to the next version.

**Review by devmil — COMMENTED** [review]
Can I learn more about your use case?
As dependency overrides only affect the local build its relevance for the public API is rather low. At least in my mind.

**Comment by mosuem** [comment]
> Can I learn more about your use case? As dependency overrides only affect the local build its relevance for the public API is rather low. At least in my mind.

We are testing a "PR health" workflow, which checks some basics about PRs and publishes the result as a comment on the workflow (see https://github.com/dart-lang/i18n/pull/710#issuecomment-1697112137). 

We were thinking about adding a nudge for people to rev the package major version on breaking API changes, so running this tool and showing its output (#137 or even better JSON output would be great for this as well btw).

IIUC, this tool copies the referenced repositories, but when adding a local dependency as a dependency override, it is not copied, so the build fails, see https://github.com/dart-lang/ecosystem/pull/153#issuecomment-1697856568.

**Comment by devmil** [comment]
> > Can I learn more about your use case? As dependency overrides only affect the local build its relevance for the public API is rather low. At least in my mind.
> 
> We are testing a "PR health" workflow, which checks some basics about PRs and publishes the result as a comment on the workflow (see [dart-lang/i18n#710 (comment)](https://github.com/dart-lang/i18n/pull/710#issuecomment-1697112137)).
> 
> We were thinking about adding a nudge for people to rev the package major version on breaking API changes, so running this tool and showing its output (formatted to align with the other outputs, so #137 or even better JSON output would be great for this as well btw).
> 
> IIUC, this tool copies the referenced repositories, but when adding a local dependency as a dependency override, it is not copied, so the build fails, see [dart-lang/ecosystem#153 (comment)](https://github.com/dart-lang/ecosystem/pull/153#issuecomment-1697856568).

Ok, thanks for the context. True, the copy process needs to respect local build options .
I totally agree with the JSON output, this would help later processing.

**Comment by mosuem** [comment]
This is still WIP, there are still some issues for example when running

`dart run bin/main.dart diff --old pub://firehose/0.3.24 --new ../ecosystem/pkgs/firehose/ -p`

with a local copy of `package:firehose` from [here](https://github.com/dart-lang/ecosystem/tree/main/pkgs/firehose).

**Comment by mosuem** [comment]
@devmil This seems to be fixed now by not running pub get on the package directly, but a dummy package, and then copying over the `package_config.json` file.

**Comment by devmil** [comment]
> @devmil This seems to be fixed now by not running pub get on the package directly, but a dummy package, and then copying over the `package_config.json` file.

Nice idea! This would make the directory structure analysis logic completely obsolete and remove all that complexity.

**Comment by mosuem** [comment]
Not my idea (@jonasfj), but I agree :)

**Comment by devmil** [comment]
wouldn't it be enough to just call pub get on the original package and copy the `package_config.json` from there?

**Pull request closed without merging**